public/app/features/dashboard/partials/dashboard_import.html

```diff
@@ -132,10 +146,10 @@ <h3 class="section-heading">
     </div>
 
     <div class="gf-form-button-row">
-      <button type="button" class="btn btn-success width-12" ng-click="ctrl.saveDashboard()" ng-hide="ctrl.nameExists || ctrl.uidExists" ng-disabled="!ctrl.inputsValid">
+      <button type="button" class="btn btn-success width-12" ng-click="ctrl.saveDashboard()" ng-hide="ctrl.nameExists || ctrl.uidExists" ng-disabled="!ctrl.isValid()">
```

Comment:
Should be *ctrl.inputsValid* so that you can import with an overwrite as long as you've selected all needed datasources.

Reply:
Idea of this is diable import if folder wasn't selected:
```js
isValid() {
  return this.inputsValid && !this.hasNameValidationError && this.folderId !== null;
}
```
But if it's possible to override dashboard with the same name (but without other name validation error), caheck should be like this I think:
```js
return this.inputsValid && 
this.folderId !== null &&
(!this.hasNameValidationError || this.nameExists)
```

Reply:
I think it's better to have a variable to check than a function from an Angular performance perspective. 

In my head the only thing you should need to check for disabled is `!ctrl.inputsValid || ctrl.folderId === null`.

Reply:
I've just checked this. `validateNewDashboardName()` returns 2 types of errors. First, if name exists. Second one - if name is empty. But if name is empty, you cannot save dashboard, so your note makes sense I think.

Reply:
About performance. Not sure variable is much faster than a short function. Anyway, angular should evaluate an expression, whether it's variable or function.